Plugins/Mute/DataManager.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Data.Models.Client;
+using SharedLibraryCore.Interfaces;
+using static System.Enum;
+
+namespace Mute;
+
+public class DataManager
+{
+    public DataManager(IMetaServiceV2 metaService)
+    {
+        _metaService = metaService;
+    }
+
+    private readonly IMetaServiceV2 _metaService;
+
+    public async Task<MuteState> ReadPersistentData(EFClient client)
+    {
+        var clientMeta = (await _metaService.GetPersistentMeta(Plugin.MuteKey, client.ClientId))?.Value ??
+                         nameof(MuteState.Unmuted);
+        
+        return Parse<MuteState>(clientMeta);
+    }
+
+    public async Task WritePersistentData(EFClient client, MuteState state)
+    {
+        await _metaService.SetPersistentMeta(Plugin.MuteKey, state.ToString(), client.ClientId);
+        client.SetAdditionalProperty(Plugin.MuteKey, state);
```

Comment:
This set of additional properties should arguably be done in the Mute manager, since based off the name of the class it should only deal with data storage/retrieval.

Reply:
I've made a few changes to read for it either to make more or less sense.